crates/sui-core/src/consensus_handler.rs

```diff
@@ -94,10 +98,23 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
     ) {
         let _scope = monitored_scope("HandleConsensusOutput");
         let mut sequenced_transactions = Vec::new();
-        let mut seq = 0;
 
         let mut bytes = 0usize;
         let round = consensus_output.sub_dag.round();
+
+        /* (serialized, transaction, output_cert) */
+        let mut transactions = vec![];
+
+        let prologue_transaction = self.consensus_commit_prologue_transaction(
```

Comment:
so does this become the first tx of every checkpoint?

Reply:
This is first transaction on every narwhal commit. Since commit can be split between multiple checkpoints, not every checkpoint will have this as first transaction

Reply:
Actually, even within checkpoint there is no special code that will put this transaction first, it will go before any transaction that reads Clock object in that checkpoint, but otherwise can be anywhere

Reply:
got it, thanks